app/controllers/api/internal/tax_documents_controller.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class Api::Internal::TaxDocumentsController < Api::Internal::BaseController
+  before_action :authenticate_user!
+  before_action :ensure_tax_center_enabled
+  after_action :verify_authorized
+
+  def index
```

Comment:
Internal API used to fetch the tax forms data for the selected year in the year dropdown on the tax center page.

Reply:
Looking at the code, it relies on an `isLoading` boolean. You would still need it, but you could set it at different events of the Inertia visit in a similar fashion than what the current code does.

https://inertiajs.com/docs/v2/the-basics/manual-visits#event-callbacks.

It would look like something like this:

```js
import { router } from '@inertiajs/react'

// [...]

router.get(Routes.tax_center_path({ year }), {
    only: ["tax_documents"],
    onStart: (_visit) => setIsLoading(true),
    onFinish: (_visit) => setIsLoading(false),
})
```

But sure, we can tackle this separately!